lib/rules/no-smart-quotes.js

```diff
@@ -41,29 +49,47 @@ module.exports = {
                     return;
                 }
 
-                const match = node.raw.match(/^(.*(“|”|‘|’).*)$/);
-
-                if (match) {
-                    const offendingPunctuation = match[2];
-                    const matchedString = match[1];
-                    const alternate = {
-                        '“': '"',
-                        '”': '"',
-                        '‘': "'",
-                        '’': "'",
-                    } [offendingPunctuation];
-
-                    if (typeof matchedString !== 'undefined' && typeof offendingPunctuation !== 'undefined') {
-                        const message = `Strings must use straight quotes in: ${ matchedString}. Use ${alternate} instead.`;
-                        context.report({
-                            node,
-                            message: message,
-                            data: {
-                                offendingPunctuation
-                            }
-                        });
-                    }
+                const matches = Array.from(matchAll(node.raw, SMART_QUOTES_REGEX));
```

Comment:
I'm now matching every smart quote in the string at once (using `matchAll`) so much of the rule has been updated around that.